client/modules/workplace/components/collection_fields/object_tree.jsx

```diff
@@ -32,17 +71,28 @@ class CollectionFields extends React.Component {
             nestLevel={nestLevel}
             updateObjectTree={this.updateObjectTree}
             updateCollectionField={updateCollectionField}
+            onClick={this.handleClick(data)}
           />
         }
         itemClassName={
           !data.nestedData ? `no-arrow nest-level-${nextLevel}` : `nest-level-${nextLevel}`
         }
-        defaultCollapsed={nestLevel > 0}
+        collapsed={data.isCollapsed}
+        onClick={this.handleClick(data)}
       >
         {data.nestedData && data.nestedData.map((item, j) => this.treeView(item, j, nextLevel))}
       </TreeView>
     );
   }
+  handleClick(field) {
+    if (field.type !== 'object') return null;
```

Comment:
lets check for `field.nestedData` field.type may be an `array` which too should be expandable.